lib/agenda/database.js

```diff
@@ -37,7 +38,8 @@ module.exports = function(url, collection, options, cb) {
       return;
     }
     debug('successful connection to MongoDB using collection: [%s]', collection);
-    self._mdb = db;
+    self._db = client;
+    self._mdb = client.db(database);
```

Comment:
I wouldn't reach in to `client.s.options.dbName`. If you do `client.db()` with no args, that gives you the database that is specified in the URI. [That's how mongoose does it](https://github.com/Automattic/mongoose/blob/fc76fe4f68ad013fcb7ca117e2539ee971a3679a/lib/connection.js#L493). Here are [the relevant docs](http://mongodb.github.io/node-mongodb-native/3.1/api/MongoClient.html#db)

Reply:
>I wouldn't reach in to `client.s.options.dbName`. If you do `client.db()` ...

Excellent! I didn't like accessing it directly like that either.